tests/test_workers.py

```diff
@@ -0,0 +1,238 @@
+from __future__ import annotations
+
+import signal
+import subprocess
+import sys
+import tempfile
+import time
+from typing import TYPE_CHECKING
+
+import httpx
+import pytest
+
+if TYPE_CHECKING:
+    from ssl import SSLContext
+    from typing import IO, Generator
+
+    from asgiref.typing import (
+        ASGIReceiveCallable,
+        ASGISendCallable,
+        LifespanStartupFailedEvent,
+        Scope,
+    )
+
+pytestmark = pytest.mark.skipif(sys.platform == "win32", reason="requires unix")
+gunicorn_arbiter = pytest.importorskip("gunicorn.arbiter", reason="requires gunicorn")
+uvicorn_workers = pytest.importorskip("uvicorn.workers", reason="requires gunicorn")
+
+
+class Process(subprocess.Popen):
+    client: httpx.Client
+    output: IO[bytes]
+
+    def read_output(self) -> str:
+        self.output.seek(0)
+        return self.output.read().decode()
+
+
+@pytest.fixture(
+    params=(
+        pytest.param(uvicorn_workers.UvicornWorker, marks=pytest.mark.subprocess),
+        pytest.param(uvicorn_workers.UvicornH11Worker, marks=pytest.mark.subprocess),
+    )
+)
+def worker_class(request: pytest.FixtureRequest) -> str:
```

Comment:
A custom marker `pytest.mark.subprocess` will be added to all tests using the subprocess fixtures. Subprocess tests can be skipped by adding `-m "not subprocess"` to the pytest command.

Reply:
It would be cool if we could do the opposite.

Reply:
See related comment https://github.com/encode/uvicorn/pull/1995#discussion_r1220727158